src/components/fuzzySearch.ts

```diff
@@ -0,0 +1,118 @@
+import Fuse from "fuse.js";
+import { WebviewApi } from "vscode-webview";
+
+export enum SearchType {
+    variables = 'variables',
+    features = 'features'
+}
+//View Provider
+export const getCustomDropdown = (optionElements: string, selectedValue: string) => {
+    return `
+    <div class="custom-dropdown">
+    <input type="text" class="dropdown-input" placeholder="Search..." value="${selectedValue}">
+    <div class="dropdown-arrow">^</div>
+    <div class="dropdown-options">
+      ${optionElements}
+    </div>
+  </div>`
+}
+
+//View
+
+/*
+ default accept an array of objects with the following type:
+ {
+    label: string,
+    value: string
+ }
+*/
+export function handleFuzzySearchDropdown(
+    vscode: WebviewApi<unknown>,
+    searchType: SearchType,
+    fuseOptions: Fuse.IFuseOptions<any> = {
+        keys: ['label', 'value'],
+        threshold: 0.5,
+    }
+) {
+    window.addEventListener('message', (event) => {
+        const message = event.data
+        if (message.type === 'searchData') {
+            const data = JSON.parse(message.value)
+            localStorage.setItem(message.searchType, JSON.stringify(data))
+        }
+    })
+
+    const localStorageData = localStorage.getItem(searchType)
+
+    if (!localStorageData) {
+        return
+    }
+
+    const input = document.querySelector('.dropdown-input') as HTMLInputElement
+    const optionsList = document.querySelector('.dropdown-options') as HTMLDivElement
+
+    let fuse: Fuse<{ label: string, value?: string }>
+    const searchData: { label: string, value?: string }[] = JSON.parse(localStorageData)
+
+    fuse = new Fuse([...searchData], fuseOptions)
+
+    const createAndAppendOption = (displayValue: string, dataAttributeValue?: string) => {
+        const option = document.createElement('div')
+        option.textContent = displayValue
+        option.setAttribute('data-value', dataAttributeValue ? dataAttributeValue : displayValue)
```

Comment:
just a nitpick, but this could be written more simply without the ternary operator
```suggestion
        option.setAttribute('data-value', dataAttributeValue || displayValue)
```